File: lang/thrift/hello_server.go
package main

import (
	"./gen-go/hello"
	"fmt"
	"git.apache.org/thrift.git/lib/go/thrift"
)

type HelloServiceImpl struct {
}

func (this *HelloServiceImpl) HelloFunc() (r string, err error) {
	fmt.Println("[Server] got request")
	return "Hello from go server", nil
}

func main() {
	handler := &HelloServiceImpl{}
	processor := hello.NewHelloServiceProcessor(handler)

	listenSocket, err := thrift.NewTServerSocket(":8787")
	if err != nil {
		panic(err)
	}

	server := thrift.NewTSimpleServer2(processor, listenSocket)
	fmt.Println("Server ready")

	if err := server.Serve(); err != nil {
		panic(err)
	}
}

File: lang/thrift/hello_server.py
import sys
sys.path.append("gen-py")

from thrift.transport import TSocket 
from thrift.server import TServer 
from hello import HelloService

class HelloHandler:
    def hello_func(self):
        print '[Server] handling client request'
        return 'hello thrift, from py server'


handler = HelloHandler()
processor = HelloService.Processor(handler)
listen_sock = TSock.TServerSocket(port=8787)
server = TServer.TSimpleServer(processor, listen_sock)

print '[Server] started'
server.serve()

